File: matiasmasso/Mat.NET/DT2/DTOSpvOut.vb
Public Class DTOSpvOut
    Property Yea As Integer
    Private pId As Integer
    Property Customer As DTOCustomer
    Property SuProveedorNum As String
    Property Fch As Date
    Property Kgs As Integer
    Property M3 As Decimal
    Property Bts As Integer
    Property PortsCod As DTOCustomer.PortsCodes
    Property Cod As Integer 'reparacions=4
    Property Cfp As Integer
    Property Spvs As List(Of DTOSpv)
    Property Usr As DTOUser
    Property Etq As String
    Property Nom As String
    Property Adr As String
    Property Cit As String
    Property Tel As String
    Property ECB As String
    Property Cash As Boolean
    Property Val As Decimal
    Property Dto As Single
    Property Dpp As Single
    Property Iva As Single
    Property Req As Single
    Property Recogeran As Boolean
    Property Delivery As DTODelivery


    Shared Function Factory(oUser As DTOUser, DtFch As Date) As DTOSpvOut
        Dim retval As New DTOSpvOut
        With retval
            .Usr = oUser
            .Fch = DtFch
        End With
        Return retval
    End Function

    Public Sub RestoreObjects()
        If _Delivery IsNot Nothing Then
            _Delivery.restoreObjects()
        End If
        For Each oSpv In _Spvs
            oSpv.restoreObjects()
        Next
    End Sub
End Class

File: matiasmasso/Mat.NET/DTO/DTOSpvOut.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTOSpvOut
    {
        public int Yea { get; set; }
        //private int pId;
        public DTOCustomer Customer { get; set; }
        public string SuProveedorNum { get; set; }
        public DateTime Fch { get; set; }
        public int Kgs { get; set; }
        public decimal M3 { get; set; }
        public int Bts { get; set; }
        public DTOCustomer.PortsCodes PortsCod { get; set; }
        public int Cod { get; set; } // reparacions=4
        public int Cfp { get; set; }
        public List<DTOSpv> Spvs { get; set; }
        public DTOUser Usr { get; set; }
        public string Etq { get; set; }
        public string Nom { get; set; }
        public string Adr { get; set; }
        public string Cit { get; set; }
        public string Tel { get; set; }
        public string ECB { get; set; }
        public bool Cash { get; set; }
        public decimal Val { get; set; }
        public float Dto { get; set; }
        public float Dpp { get; set; }
        public float Iva { get; set; }
        public float Req { get; set; }
        public bool Recogeran { get; set; }
        public DTODelivery Delivery { get; set; }


        public static DTOSpvOut Factory(DTOUser oUser, DateTime DtFch)
        {
            DTOSpvOut retval = new DTOSpvOut();
            {
                var withBlock = retval;
                withBlock.Usr = oUser;
                withBlock.Fch = DtFch;
            }
            return retval;
        }

        public void RestoreObjects()
        {
            if (Delivery != null)
                Delivery.RestoreObjects();
            foreach (var oSpv in Spvs)
                oSpv.restoreObjects();
        }
    }
}
